secure_tunneling/source/SecureTunnel.cpp

```diff
@@ -10,6 +10,189 @@ namespace Aws
 {
     namespace Iotsecuretunneling
     {
+        void setPacketByteBufOptional(
+            Crt::Optional<Crt::ByteCursor> &optional,
+            Crt::ByteBuf &optionalStorage,
+            Crt::Allocator *allocator,
+            const Crt::ByteCursor *value)
+        {
+            aws_byte_buf_clean_up(&optionalStorage);
+            AWS_ZERO_STRUCT(optionalStorage);
+            if (value != nullptr)
+            {
+                aws_byte_buf_init_copy_from_cursor(&optionalStorage, allocator, *value);
+                optional = aws_byte_cursor_from_buf(&optionalStorage);
+            }
+            else
+            {
+                optional.reset();
+            }
+        }
+
+        void setPacketStringOptional(
+            Crt::Optional<Crt::ByteCursor> &optional,
+            Crt::String &optionalStorage,
+            const aws_byte_cursor *value)
+        {
+            if (value != nullptr)
+            {
+                optionalStorage = Crt::String((const char *)value->ptr, value->len);
+                struct aws_byte_cursor optional_cursor;
+                optional_cursor.ptr = (uint8_t *)optionalStorage.c_str();
+                optional_cursor.len = optionalStorage.size();
+                optional = optional_cursor;
+            }
+        }
+
+        //***********************************************************************************************************************
+        /*                                              Message */
+        //***********************************************************************************************************************
+
+        Message::Message(const aws_secure_tunnel_message_view &message, Crt::Allocator *allocator) noexcept
+            : m_allocator(allocator)
+        {
+            AWS_ZERO_STRUCT(m_payload);
```

Comment:
I'm not sure if you can do a AWS_ZERO_STRUCT to a Crt::Optional. Probably need a further testing on it...

Reply:
Cool! Remove the Zero Struct for both optional members.